File: Front/TS/src/infrastructure/adapters/gateways/HttpUserGateway.ts
import { IUserGateway } from "../../../domain/abstractions/IUserGateway";
import { IHttpService } from "../http/IHttpService";

export class HttpUserGateway implements IUserGateway {
    constructor(private readonly httpService: IHttpService){}

    async Register(user: { id: string; username: string; email: string; password: string; }): Promise<void> {
        const response = await this.httpService.Post("/users", user)
        if(response.Code == 200) {
            return Promise.resolve()
        }
        return Promise.reject(response.Error);
    }

}

File: Front/.Net Maui/BetFriends.Infrastructure/Gateways/Http/HttpUserGateway.cs
using BetFriends.Domain.Abstractions;
using System.Net.Http.Json;

namespace BetFriends.Infrastructure.Gateways.Http;

internal class HttpUserGateway(IHttpClientFactory httpClientFactory) : IUserGateway
{
    private readonly HttpClient httpClient = httpClientFactory.CreateClient("authentication");
    public async Task SaveAsync(User user)
    {
        var response = await httpClient.PostAsJsonAsync("/users", new
        {
            user.Username,
            user.Email,
            user.Id,
            user.Password
        });
        response.EnsureSuccessStatusCode();
    }
}
